src/main/java/org/openrewrite/java/dropwizard/annotation/AddClassAnnotationRecipe.java

```diff
@@ -0,0 +1,102 @@
+import static org.openrewrite.java.JavaParser.runtimeClasspath;
+
+public abstract class AddClassAnnotationRecipe extends Recipe {
+    private final String annotationText;
+    private final Boolean annotateSubclasses;
+
+    public AddClassAnnotationRecipe(String annotationText, Boolean annotateSubclasses) {
+        this.annotationText = annotationText;
+        this.annotateSubclasses = annotateSubclasses;
+    }
+
+    protected abstract boolean shouldAddAnnotation(ClassDeclaration cd);
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public ClassDeclaration visitClassDeclaration(
+                    ClassDeclaration classDeclaration, ExecutionContext ctx) {
+                ClassDeclaration cd = super.visitClassDeclaration(classDeclaration, ctx);
+
+                // if annotation parameters are given as well
+                String annotationType = annotationText.split("[(<]")[0].trim();
+
+                JavaType.FullyQualified fqn = TypeUtils.asFullyQualified(JavaType.buildType(annotationType));
+
+                if (fqn == null) {
+                    return cd;
+                }
+
+                if (service(AnnotationService.class)
+                        .matches(getCursor(), new AnnotationMatcher(fqn.getFullyQualifiedName()))) {
+                    return cd;
+                }
+
+                boolean shouldAdd = shouldAddAnnotation(cd);
+                if (!shouldAdd && TRUE.equals(annotateSubclasses)) {
+                    shouldAdd = shouldAddAnnotationToAnyParentClass();
+                }
+                if (!shouldAdd) {
+                    return cd;
+                }
+                maybeAddImport(fqn);
+
+                ClassDeclaration updated =
+                        JavaTemplate.builder("@#{}")
+                                .javaParser(fromJavaVersion().classpath(runtimeClasspath()))
+                                .imports(fqn.getFullyQualifiedName())
+                                .build()
+                                .apply(
+                                        updateCursor(cd),
+                                        cd.getCoordinates().addAnnotation(comparing(J.Annotation::getSimpleName)),
+                                        annotationText);
+                doAfterVisit(service(ImportService.class).shortenFullyQualifiedTypeReferencesIn(updated));
+                return maybeAutoFormat(cd, updated, ctx);
+            }
+
+            private boolean shouldAddAnnotationToAnyParentClass() {
+                getCursor().dropParentUntil(
+                        x -> {
+                            if (x instanceof J.ClassDeclaration) {
+                                return shouldAddAnnotation((J.ClassDeclaration) x);
+                            }
+                            return false;
+                        });
+                return true;
```

Comment:
This puzzled me a bit; the response from dropParentUntil isn't used, so then this always `return true`, right? Was that intentional?